sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -329,6 +329,14 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       Row(3, "c") ::
       Row(4, "d") :: Nil)
     checkAnswer(lowerCaseData.intersect(upperCaseData), Nil)
+
+    // check null equality
```

Comment:
A good case might be two tables that each contain two int rows [1], [1] and intersect them.